apps/api/src/app/notification-template/usecases/get-notification-templates/get-notification-templates.usecase.ts

```diff
@@ -1,14 +1,22 @@
 import { Injectable } from '@nestjs/common';
 import { NotificationTemplateEntity, NotificationTemplateRepository, OrganizationEntity } from '@novu/dal';
 import { GetNotificationTemplatesCommand } from './get-notification-templates.command';
-
+import { NotificationTemplatesResponseDto } from '../../dto/notification-templates.response.dto';
 @Injectable()
 export class GetNotificationTemplates {
   constructor(private notificationTemplateRepository: NotificationTemplateRepository) {}
 
-  async execute(command: GetNotificationTemplatesCommand): Promise<NotificationTemplateEntity[]> {
-    const list = await this.notificationTemplateRepository.getList(command.organizationId, command.environmentId);
+  async execute(command: GetNotificationTemplatesCommand): Promise<NotificationTemplatesResponseDto> {
+    const LIMIT = 10;
```

Comment:
This limit could be passed from the API User with a default of 10 and maximum at 100, you can validate using class-validators on the API usecase. if you could also verify in other places of the application if we use this usecase aswell in case we do some "client-side" search on the server